distributed/tests/test_scheduler.py

```diff
@@ -225,6 +225,21 @@ def test_remove_worker_by_name_from_scheduler(s, a, b):
     s.validate_state()
 
 
+@gen_cluster()
+def test_remove_worker_increments_suspicious(s, a, b):
+    dsk = {("x-%d" % i): (inc, i) for i in range(20)}
+    s.update_graph(
+        tasks=valmap(dumps_task, dsk),
+        keys=list(dsk),
+        dependencies={k: set() for k in dsk},
+    )
+    assert s.remove_worker(address=a.address) == "OK"
```

Comment:
Ah, I see, you're using internal API in order to simulate a failing task.  

Instead, you might consider using a function that generally causes a worker to fail, like `sys.exit` (you'll want to make sure that you use `gen_cluster(Worker=Nanny)` so that we use separate worker processes.